dts/arm/nxp/nxp_lpc54xxx.dtsi

```diff
@@ -58,9 +62,20 @@
 			reg = <0x4000000 0x8000>;
 		};
 
-		flash0:flash@0 {
-			compatible = "soc-nv-flash";
-			reg = <0 0x40000>;
+		iap: flash-controller@4009c000 {
+			compatible = "nxp,lpc-iap";
+			label = "FLASH_IAP";
+			reg = <0x4009c000 0x18>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 0x40000>;
```

Comment:
I'm working to replace memory sizes with DT_SIZE_K macros in #29420. If that PR gets merged first, please update here accordingly

Reply:
i've changed them just for the flash node, so you don't have merge conflicts.

Reply:
Thanks. The buildkite check is failing because you didn't also add `#include <mem.h>`

Reply:
I've added `#include <mem.h>`